src/inspector_socket_server.cc

```diff
@@ -376,11 +385,25 @@ void InspectorSocketServer::SendListResponse(InspectorSocket* socket) {
 
 bool InspectorSocketServer::Start() {
   CHECK_EQ(state_, ServerState::kNew);
-  sockaddr_in addr;
+  struct addrinfo hints;
+  memset(&hints, 0, sizeof(struct addrinfo));
+  hints.ai_flags = AI_NUMERICSERV;
+  hints.ai_socktype = SOCK_STREAM;
+  uv_getaddrinfo_t req;
+  const std::string port_string = std::to_string(port_);
+  int err = uv_getaddrinfo(loop_, &req, nullptr, host_.c_str(),
+                           port_string.c_str(), &hints);
+  if (err < 0) {
+    if (out_ != NULL) {
+      fprintf(out_, "Unable to resolve \"%s\": %s\n", host_.c_str(),
+              uv_strerror(err));
+    }
+    return false;
+  }
   uv_tcp_init(loop_, &server_);
-  uv_ip4_addr(host_.c_str(), port_, &addr);
-  int err = uv_tcp_bind(&server_,
-                        reinterpret_cast<const struct sockaddr*>(&addr), 0);
+  err = uv_tcp_bind(&server_, req.addrinfo->ai_addr, 0);
```

Comment:
Now this class works with a vector of sockets. This should also help if there's ever a need to support  listening on multiple ports (e.g. someone mentioned before something about changing the port at the runtime).

One issue this implementation is that different sockets will bind to different ports if requested port is 0 - while only first socket port is surfaced in API and console messages. I did not want to further complicate this implementation at this time, but an API that provides list of addresses and ports can be created later.